dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -925,7 +930,7 @@ RtpsUdpDataLink::stop_i()
   }
 
   nack_reply_.cancel();
-  heartbeat_reply_.cancel();
+  heartbeat_reply_.disable_and_wait();
```

Comment:
This should be `disable()`.  MultiTask (like the others) has a weak pointer to the delegate so you can disable it without waiting.

Reply:
Hmm, all the other ones in stop_i() are also disable_and_wait()... I would expect us to want to wait in case shutdown finishes and MultiTask (PmfMultiTask, in this case) gets erased before the cancel command runs?

Reply:
I my opinion, it would be better to dynamically allocate all of the *Tasks including heartbeat_reply_ so you can take advantage of the weak pointer and not wait during shutdown.    So, you are correct that you will need to `disable_and_wait()` unless you dynamically allocate `heartbeat_replay_`.

Reply:
I looked into this a little further and the *Tasks don't support what I was claiming.  They could be modified to support it.  I'll discuss that with you in a different venue.